homeassistant/components/totalconnect/__init__.py

```diff
@@ -24,30 +27,64 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-TOTALCONNECT_PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up from existing/saved configuration."""
+    conf = config.get(DOMAIN)
+    if conf is None:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+        )
+    )
+
+    return True
 
 
-def setup(hass, config):
-    """Set up TotalConnect component."""
-    conf = config[DOMAIN]
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up upon config entry in user interface."""
+    hass.data.setdefault(DOMAIN, {})
 
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
+    conf = entry.data
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
 
     client = TotalConnectClient.TotalConnectClient(username, password)
```

Comment:
This is doing I/O, which is not allowed in async context

Reply:
Think this is fixed now.